### DISCUSSION TITLE
Update from 2.5.10 to 2.11.7

### DISCUSSION THREAD

**Discussion by jonatangobbato** [netbox-community/netbox#6683] [Q&A] [answered]
Hi NetBox community,

In my enviroment, we use netbox, but after its installation no more further updates are performed, now we are using v2.5.10 and we want to upgrade to v2.11.7. (So many improvements)

The doubt is, simply following the Upgrade documentation do the trick? (https://netbox.readthedocs.io/en/stable/installation/upgrading/)

So many versions of difference may be a problem? 

Any tips for this upgrade proccess?

Thanks

**Comment by candlerb** [comment, marked as answer]
Direct one-step upgrade generally works fine.

However, you need to read the [release notes](https://netbox.readthedocs.io/en/stable/release-notes/version-2.6/) for all the intervening major versions - in your case 2.6, 2.7, 2.8, 2.9, 2.10, 2.11 - and scroll down to the bottom for the first release ([2.6.0](https://netbox.readthedocs.io/en/stable/release-notes/version-2.6/#changes), [2.7.0](https://netbox.readthedocs.io/en/stable/release-notes/version-2.7/#changes) etc) where the breaking changes are listed.

Compare your `configuration.py` with the new `configuration.example.py` and look for any changes in settings.  Usually, new settings can be omitted, however there are some changes which need to be made.  These are called out in the release notes.  From memory, important ones are:
* changes to the redis settings
* changes to external authentication if you're using it

The REST API and the ORM API are **not** stable, and do change between versions in incompatible ways (e.g. RackGroup has changed to Location).  If you have any integrations with the REST API, any reports, custom scripts etc, they may need changing to work with the new version.

As always, the safest approach is to do the upgrade in a test environment first.  You can do this by [replicating](https://netbox.readthedocs.io/en/stable/administration/replicating-netbox/) your existing database, copying it into a fresh 2.11.7 installation, running `./upgrade.sh`, and then testing.
